### ISSUE TITLE
Identity theory

### ISSUE THREAD

**Issue by digitalic** [space-wizards/space-station-14#103]
The ~~100th~~ 103rd issue/PR of the content repo.

TL;DR at bottom

### First principles of identity in SS13/SS14

**The psyche behind the screen**

Identifying and understanding the other is a vital part of mastering any game with strategy in it. Games like EpicMafia, Town of Salem, and Trouble in Terrorist Town provide a path to mastery in understanding the mind of those you play with to analyse and predict their actions. SS13, in its PvP gamemodes involving secret information, can theoretically play the same way. But roleplay-generated narrative, immersion, fairness and sportsmanship, among the many other things that make SS13 fun and unique, conflict with this with the player as they instead require interaction with the layer below them, their "character".

The optimal way to play is still identifying and understanding your IRL opponent though, because their character is just an extension of them. Hence there are expected to be mechanics, rules, and community pushback against trying to directly gain information about other _players_ while playing. Instead, the end goal of anyone playing with the identity system is to identify character personalities. These are the closest in-character analogue to the player themselves, having a consistent (player controlled) personality and (player-remembered) memories.

**Characters and identity tells**

Thematically, a character personality is resultant of the physical state of their circuitry: the brain/robotics/ectoplasm, etc. With full knowledge of this circuitry, one could completely predict a character's actions, and reconstruct them. This isn't possible in SS14, as that "circuitry" is just a player using their input system. NPCs with AI could certainly have cloneable personalities though.

All possible character interactions with the world are, for identity purposes, a form of communication (which we will call "identity tells"). Speech and emotes are obviously communication, but so are actions, just more indirectly - such as putting on a change of clothes, sticking oneself with an injector of another's DNA, or leaving a trail of blood between themselves and the body they just proboscified. 

**Disguise and probability**

Identifying others through identity tells is a probabilistic game, as all identity tells have a probability of being imitated. The hardest tells to imitate are direct communications from personalities. An extreme example would be your character giving out a public key at the start of the round and then signing messages with their private key to verify themselves as the round goes on. Similarly, having unique quirks, mannerisms, and speech patterns can be very difficult to imitate. Bodily organs, clothes and ID, and other miscellaneous actions are all difficult to imitate depending on time, skill, resources, etc. required to do so. 

Tells have another layer to them from the perceiving side: the time, skill, and resources required to observe them. For example, clothing and an ID card are easily spoofed, but also easily observed with just some eyeballs. Conversations with a personality or DNA testing of their blood are much more solid evidence, but require more time, skill and equipment to observe.

Taken as a whole, identity tells all contribute to engage players in the identification game, where the ideal play is to expend the least amount of resources for the most accuracy in identifying the person in front of them.

**Previous relationships**

In a naive approximation of the game, personalities are first identified during the round; your character has no knowledge of the other persons on the station, so every time a bundle of identity tells interacts with him, he will either assign them a new personality or try to match them with a previous one he's encountered. 

In practice, players tend to play the same personalities, choose the same customization options, and act the same way across rounds - a practice very hard to stop for various reasons. ICly, this can be excused as knowing a character from "before the shift". In effect, any character will know a lot of persons before the round has started, which changes the identity calculus: there is a set of previously experienced personalities at the start that are very likely to correspond to their identity tells, which drops off as the round goes on and identity tells get mixed up.

**Databases**

Personalities and tells are not restricted to the memories of players and their characters; they can be stored in the game world as well. The crew manifest is the most powerful example of this, assigning names (and by extension, the associated face) to roles and ID cards. There is also the security and medical databases which store other identity tells such as pictures, blood type, and DNA. 

Such databases can skew the balance of tells in various ways depending on their factors, so it's important to balance them right. For example, in current SS13, it can be hard to get away with pretending to be a part of the crew due to the lack of corresponding entries in the manifest or security databases. Of course, the databases themselves are, in a very indirect fashion, identity tells. And if mutable in-game, they can be used for disguise.

### Takeaways (the TL;DR)

- Players who want to win are going to want to try to find out who the player they're interacting with is. For RP reasons, don't let this happen. Have characters stand in as the proxies for players.
- Everything that can be affected by a character's actions is, in some way, an identity tell. The probabilistic matching of tells to characters can be a fun and thematic part of the game. There are various factors to these tells to be used in balancing: broadly, difficulty to imitate and difficulty to observe.
- Because characters and their tells persist across rounds (hard and unpleasant to stop), and player memories of those characters persist across rounds (good luck stopping that) you need to accept that IC, characters are going to know each other across rounds. This affects the balance of the game.
- IC databases are a balancing component of the identity game, especially if they can be subverted.

### More in comments later

**Comment by digitalic** [comment]
Closing to merge into hackmd docs
